core/src/apps/nem/validators.py

```diff
@@ -106,71 +94,43 @@ def _validate_common(common: NEMTransactionCommon, inner: bool = False):
+def _validate_public_key(public_key: bytes | None, err_msg: str) -> None:
     if not public_key:
         raise ProcessError(f"{err_msg} (none provided)")
     if len(public_key) != NEM_PUBLIC_KEY_SIZE:
         raise ProcessError(f"{err_msg} (invalid length)")
 
 
-def _validate_importance_transfer(importance_transfer: NEMImportanceTransfer):
-    if importance_transfer.mode is None:
-        raise ProcessError("No mode provided")
+def _validate_importance_transfer(importance_transfer: NEMImportanceTransfer) -> None:
     _validate_public_key(
         importance_transfer.public_key, "Invalid remote account public key provided"
     )
 
 
-def _validate_multisig(multisig: NEMTransactionCommon, network: int):
+def _validate_multisig(multisig: NEMTransactionCommon, network: int) -> None:
     if multisig.network != network:
         raise ProcessError("Inner transaction network is different")
     _validate_public_key(multisig.signer, "Invalid multisig signer public key provided")
 
 
 def _validate_aggregate_modification(
     aggregate_modification: NEMAggregateModification, creation: bool = False
-):
+) -> None:
 
     if creation and not aggregate_modification.modifications:
         raise ProcessError("No modifications provided")
 
     for m in aggregate_modification.modifications:
-        if not m.type:
-            raise ProcessError("No modification type provided")
-        if m.type not in (
-            NEMModificationType.CosignatoryModification_Add,
-            NEMModificationType.CosignatoryModification_Delete,
-        ):
-            raise ProcessError("Unknown aggregate modification")
         if creation and m.type == NEMModificationType.CosignatoryModification_Delete:
             raise ProcessError("Cannot remove cosignatory when converting account")
         _validate_public_key(m.public_key, "Invalid cosignatory public key provided")
 
 
-def _validate_supply_change(supply_change: NEMMosaicSupplyChange):
-    if supply_change.namespace is None:
-        raise ProcessError("No namespace provided")
-    if supply_change.mosaic is None:
-        raise ProcessError("No mosaic provided")
-    if supply_change.type is None:
-        raise ProcessError("No type provided")
-    elif supply_change.type not in [
-        NEMSupplyChangeType.SupplyChange_Decrease,
-        NEMSupplyChangeType.SupplyChange_Increase,
-    ]:
-        raise ProcessError("Invalid supply change type")
-    if supply_change.delta is None:
-        raise ProcessError("No delta provided")
-
-
-def _validate_mosaic_creation(mosaic_creation: NEMMosaicCreation, network: int):
-    if mosaic_creation.definition is None:
-        raise ProcessError("No mosaic definition provided")
-    if mosaic_creation.sink is None:
-        raise ProcessError("No creation sink provided")
-    if mosaic_creation.fee is None:
-        raise ProcessError("No creation sink fee provided")
+def _validate_supply_change(supply_change: NEMMosaicSupplyChange) -> None:
```

Comment:
i'm keeping it for consistency -- when reading `validate()`, all variants seem covered.

but it's a matter of taste, we could as well leave it out.

it was safe to delete the checks because the function was checking (a) "is a required field present?" and (b) "is an enum field one of the enum values?", both of which protobuf codec does for us now